### ISSUE TITLE
Python 3 - TypeError: 'cmp' is an invalid keyword argument for this function

### ISSUE THREAD

**Issue by reubenfarrelly** [kdeldycke/maildir-deduplicate#34]
This is with:
```
thunderstorm murray # python --version
Python 3.5.1
```

```
thunderstorm murray # mdedup --version
mdedup, version 1.2.0
```

The system is a Gentoo Linux x86_64 system.

I am testing with a small Maildir which I know has some duplicates in it, this is what I'm seeing when I run mdedup:

```python
thunderstorm murray # mdedup -v  DEBUG deduplicate -n Maildir
Processing 35 mails in /home/murray/Maildir
debug: Hash is 2432b264412f83b743f76820494027b6920b6d5822d6c6d2a472d7bc for mail '1466129749.M534299P7950.thunderstorm,S=503405,W=509974'.
debug: Hash is 03b7105677137585e934bf43b7de418b2f9638ed2efe1a3da6029e32 for mail '1466129749.M534325P7950.thunderstorm,S=5053,W=5142'.
debug: Hash is a258a1c6dc1204b907e381169cd1ad0dec65c77a36886f4184933f10 for mail '1466129749.M534304P7950.thunderstorm,S=14792,W=15108'.
debug: Hash is 28ad5b710b388667b0bf0ee847e2cf7174e15f5d823d566d68256dda for mail '1466129749.M534306P7950.thunderstorm,S=8607,W=8732'.
debug: Hash is 624d4ea689096d26b3de13bf3f54dd2f02635c5a9b79797af3978d97 for mail '1466129749.M534295P7950.thunderstorm,S=41767,W=42408'.
debug: Hash is 2fb8ad807a3197cf6837551ca428dadb74821a9b93f96e2eb6415ec5 for mail '1466129749.M534307P7950.thunderstorm,S=26740,W=27240'.
debug: Hash is b720bef3e4b27ce0f703ee13941532478e24d9e713b07d9903ab3a05 for mail '1466129749.M534293P7950.thunderstorm,S=5898,W=6043'.
debug: Hash is 16192e2336bf9919991257e3b4d00de7eca399d6e69f3ef6dd84673b for mail '1466129749.M534298P7950.thunderstorm,S=42484,W=43049'.
debug: Hash is 3160628605746bc50097d8742379f3a5c3ab6e8dd589d3c2eddffa31 for mail '1466129749.M534291P7950.thunderstorm,S=973277,W=985987'.
debug: Hash is e18bcdc38cb10701ec0a56eb85f6d8383e27abcb67b4bd3f09c3c48d for mail '1466129749.M534294P7950.thunderstorm,S=4654,W=4716'.
debug: Hash is b011ffbfd46f40a349979b05347c6eb51f84ed38e8b5ea8d4908c886 for mail '1466129749.M534303P7950.thunderstorm,S=48249,W=48921'.
debug: Hash is d3f39b687a4eebfa17ddaedda8eb39736d623748f956313a7d9a5d30 for mail '1466129749.M534292P7950.thunderstorm,S=338951,W=343443'.
debug: Hash is 5033048eaeec5b379e9c843aa0e0ce57e031fbc647b22a627544766a for mail '1466129749.M534316P7950.thunderstorm,S=9215,W=9358'.
debug: Hash is 6525592585c5e47b62383082f213af2c714ab1dd7d8b0ae607eaa733 for mail '1466129749.M534323P7950.thunderstorm,S=119429,W=121141'.
debug: Hash is 5549def4c02ea0e1210f002e967781a3bc8c2bece635f004336abc89 for mail '1466129749.M534297P7950.thunderstorm,S=38415,W=39209'.
debug: Hash is 664e868121a40651be43b054c6c8a0bf2b3a46bdfa17fa6fc1a63c77 for mail '1466129749.M534311P7950.thunderstorm,S=1112515,W=1126976'.
debug: Hash is 748772fb4e7a3d32acaf47f40496d73867aa41fa41c386001625e1e2 for mail '1466129749.M534296P7950.thunderstorm,S=101196,W=102544'.
debug: Hash is 7fedb8343a6735bbdda8d01d208e6204497a41f4b219788d70544f81 for mail '1466129749.M534319P7950.thunderstorm,S=3741134,W=3789766'.
debug: Hash is 5033048eaeec5b379e9c843aa0e0ce57e031fbc647b22a627544766a for mail '1466129749.M534317P7950.thunderstorm,S=7193,W=7310'.
debug: Hash is 79ff9e78a43a8a4c54eae798d64c6fc66d2cc4c67677f38816fb0929 for mail '1466129749.M534301P7950.thunderstorm,S=25528,W=25993'.
debug: Hash is 8a82df1dbd771a67bb7ee49c202a3578b30a82012dc952329962779b for mail '1466129749.M534300P7950.thunderstorm,S=5827,W=5925'.
debug: Hash is 3d678c15b79cc82e08ded13806b9a46a85df10f8f8a0fd140a51fca8 for mail '1466129749.M534305P7950.thunderstorm,S=28634,W=29143'.
debug: Hash is 4db378bed3f7b340f7d04234991504cf113cb1c91bc4a568fe412994 for mail '1466129749.M534308P7950.thunderstorm,S=50835,W=51542'.
debug: Hash is fce21633281251a5e7a697fde97e52ba770ff30b0439abdf1ac1b8a1 for mail '1466129749.M534302P7950.thunderstorm,S=33100,W=33558'.
debug: Hash is 0f781e5d7632351335e2386c68535b0ad1e7120e065c60ac0bf65516 for mail '1466129749.M534321P7950.thunderstorm,S=7406134,W=7502437'.
debug: Hash is 23e8315bb0f1cf1d29c1987efe7cf53a32fbb7338fb89d34073de595 for mail '1466129749.M534324P7950.thunderstorm,S=8175924,W=8282150'.
debug: Hash is 83afd5fc7a59445061f87b4e1ec793ca7452cf13696437fa9ce09a1b for mail '1466129749.M534314P7950.thunderstorm,S=5896,W=5995'.
debug: Hash is 84c6f6e223fec5a208aaab354e1b022ce0443cddf1c6651597f84f04 for mail '1466129749.M534322P7950.thunderstorm,S=9247928,W=9368076'.
debug: Hash is 1ae2e0a13bd557ffd75a7610c7003a109d60fbfb14701d72525cf5ad for mail '1466129749.M534309P7950.thunderstorm,S=36576,W=37079'.
debug: Hash is 664e868121a40651be43b054c6c8a0bf2b3a46bdfa17fa6fc1a63c77 for mail '1466129749.M534313P7950.thunderstorm,S=445424,W=451222'.
debug: Hash is 8433ac85e3449f0611e40dcaeb4e9465b57ae2c0b9a8c8327579fb2f for mail '1466129749.M534318P7950.thunderstorm,S=29628,W=30231'.
debug: Hash is 7fedb8343a6735bbdda8d01d208e6204497a41f4b219788d70544f81 for mail '1466129749.M534320P7950.thunderstorm,S=3741134,W=3789766'.
debug: Hash is 5033048eaeec5b379e9c843aa0e0ce57e031fbc647b22a627544766a for mail '1466129749.M534315P7950.thunderstorm,S=11217,W=11386'.
debug: Hash is 664e868121a40651be43b054c6c8a0bf2b3a46bdfa17fa6fc1a63c77 for mail '1466129749.M534310P7950.thunderstorm,S=620741,W=628816'.
debug: Hash is 664e868121a40651be43b054c6c8a0bf2b3a46bdfa17fa6fc1a63c77 for mail '1466129749.M534312P7950.thunderstorm,S=469343,W=475452'.
Subject: Prayers of the faithfull
Traceback (most recent call last):
  File "/usr/bin/mdedup", line 9, in <module>
    load_entry_point('maildir-deduplicate==1.2.0', 'console_scripts', 'mdedup')()
  File "/usr/lib64/python3.5/site-packages/click/core.py", line 716, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib64/python3.5/site-packages/click/core.py", line 696, in main
    rv = self.invoke(ctx)
  File "/usr/lib64/python3.5/site-packages/click/core.py", line 1060, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/usr/lib64/python3.5/site-packages/click/core.py", line 889, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib64/python3.5/site-packages/click/core.py", line 534, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib64/python3.5/site-packages/click/decorators.py", line 17, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/usr/lib64/python3.5/site-packages/maildir_deduplicate/cli.py", line 146, in deduplicate
    dedup.run()
  File "/usr/lib64/python3.5/site-packages/maildir_deduplicate/deduplicate.py", line 229, in run
    sorted_messages_size = self.size_sort(messages)
  File "/usr/lib64/python3.5/site-packages/maildir_deduplicate/deduplicate.py", line 343, in size_sort
    sizes.sort(cmp=_sort_by_size)
TypeError: 'cmp' is an invalid keyword argument for this function
thunderstorm murray # 
```

**Comment by kdeldycke** [comment]
That's embarrassing. `maildir-deduplicate` really needs proper unittests... In the mean time, if you manage to produce a little fix, I'll merge it.

**Comment by jakobs** [comment]
I get the same error:
Python 3.5.2
mdedup, version 1.2.0

**Comment by jakobs** [comment]
Seems to be a problem with python3:
http://code.activestate.com/recipes/576653-convert-a-cmp-function-to-a-key-function/

**Comment by kdeldycke** [comment]
Fixed in c272a65f796cd9183f8c29662c75955155773781 .
